scope/src/doctor/check.rs

```diff
@@ -1040,4 +1045,143 @@ pub(crate) mod tests {
             .await;
         assert!(res.is_ok());
     }
+
+    #[tokio::test]
+    async fn test_glob_walker_update_path_honors_tilde_paths() {
+        let mut file_system = MockFileSystem::new();
+        let mut file_cache = MockFileCache::new();
+
+        let user_dirs = UserDirs::new().expect("coudn't initialize UserDirs");
+        let home_dir = user_dirs.home_dir().to_owned();
+        let resolved_path = home_dir.join("path/foo.txt").as_path().to_owned();
+
+        file_cache
+            .expect_update_cache_entry()
+            .once()
+            .with(
+                predicate::eq("group_name".to_string()),
+                predicate::eq(resolved_path.clone()),
+            )
+            .returning(|_, _| Ok(()));
+
+        file_system
+            .expect_find_files()
+            .once()
+            .withf(move |glob_pattern| {
+                home_dir
+                    .join("path/*.txt")
+                    .as_path()
+                    .to_owned()
+                    .to_str()
+                    .unwrap()
+                    == glob_pattern
```

Comment:
This was painful. Would it be easier if we changed the `make_absolute` signature take a `PathBuf` for glob instead of a string?